include/dronecore.h

```diff
@@ -30,17 +59,83 @@ class DroneCore
         CONNECTIONS_EXHAUSTED
     };
 
+    /**
+     * Translates the ConnectionResult enum to a human-readable English string.
+     */
     static const char *connection_result_str(ConnectionResult);
 
+    /**
+     * Adds a UDP connection with the default arguments.
+     *
+     * This will listen on UDP port 14540 on localhost.
+     *
+     * @return the result of adding the connection.
+     */
     ConnectionResult add_udp_connection();
+
+    /**
+     * Adds a UDP connection with a specific port number.
+     *
+     * @param local_port_number The local UDP port to listen to
+     * @return the result of adding the connection.
+     */
     ConnectionResult add_udp_connection(int local_port_number);
 
+    /**
+     * Return vector of device UUIDs.
+     *
+     * This returns a vector of the UUIS of all devices that have been discovered so far and can be
+     * accessed.
+     *
+     * @return a reference to the vector containing the UUIDs
+     */
     const std::vector<uint64_t> &device_uuids() const;
+
+    /**
+     * Return the only discovered device.
+     *
+     * This returns the only discovered device or a null device if no device has been discovered
+     * yet. If there is more than one device discovered, it will just return the device which was
+     * discovered first.
+     *
+     * @return a reference to a device
+     */
     Device &device() const;
+
+    /**
+     * Return a device given a UUID.
+     *
+     * This returns a device for a given UUID if such a device has been discovered and a null
+     * device otherwise.
+     */
     Device &device(uint64_t uuid) const;
 
+    /**
+     * Callback type for discover and timeout notifications.
+     *
+     * @param uuid UUID of device
+     */
     typedef std::function<void(uint64_t uuid)> event_callback_t;
+
+    /**
+     * Register callback for device discovery.
+     *
+     * This sets a callback that will be notified if a new device is discovered.
+     *
+     * Note that only one callback can be registered at a time. If this function is called several
+     * times, previous callbacks will be overwritten.
+     */
     void register_on_discover(event_callback_t callback);
+
+    /**
+     * Register callback for device timeout.
+     *
+     * This sets a callback that will be notified if  device has timed out which means that no
```

Comment:
Extra space between `if  device`

Reply:
Perhaps:
```
This sets a callback that will be notified if there is a timeout while listening for device connections. The timeout occurs if no messages are received in 3 seconds.
```
This is a little unclear - what messages are we waiting on - connection messages, heartbeats?

Reply:
It is in fact heartbeats but I wanted to hide that from the user of the API. It should not be something to be concerned about. On the other hand, a heartbeat is clear enough and does not just apply to mavlink.

Reply:
Changed in 8327f85.